### PR TITLE
SAN 3876 emit disconnected event

### PR DESCRIPTION
**Pull request by podviaznikov**
Emit disconnected event that mavis can pickup and also sauron will pickup in the future to remove the host from the network

### CODE REVIEW

File: lib/listener.js [hunk 1]
```diff
@@ -63,11 +63,11 @@ module.exports = class EventListener {
 
       this.eventStream = eventStream
       this.eventStream
-        .on('error', (e) => this.handleError(e))
-        .on('close', () => this.handleClose(new Error('stream closed')))
-        .on('end', () => this.handleClose(new Error('stream ended')))
-        .on('disconnect', () => this.handleClose(new Error('stream disconnected')))
-        .on('data', (d) => this.publishEvent(d))
+        .on('error', this.handleError.bind(this))
```
Comment (anandkumarpatel) [inline lib/listener.js:66]: :-1: not blocking though

File: lib/listener.js [hunk 2]
```diff
@@ -63,11 +63,11 @@ module.exports = class EventListener {
...
+        .on('close', this.handleClose.bind(this, new Error('stream closed')))
```
Comment (anandkumarpatel) [inline lib/listener.js:67]: :boom:

File: lib/listener.js [hunk 3]
```diff
@@ -63,11 +63,11 @@ module.exports = class EventListener {
...
+        .on('end', this.handleClose.bind(this, new Error('stream ended')))
```
Comment (anandkumarpatel) [inline lib/listener.js:68]: :bomb:

File: lib/listener.js [hunk 4]
```diff
@@ -63,11 +63,11 @@ module.exports = class EventListener {
...
+        .on('disconnect', this.handleClose.bind(this, new Error('stream disconnected')))
```
Comment (anandkumarpatel) [inline lib/listener.js:69]: :cry:

File: lib/listener.js [hunk 5]
```diff
@@ -63,11 +63,11 @@ module.exports = class EventListener {
...
+        .on('data', this.publishEvent.bind(this))
```
Comment (anandkumarpatel) [inline lib/listener.js:70]: :crying_cat_face:

**Comment by anandkumarpatel** [comment]
+1

File: lib/rabbitmq.js [hunk 6]
```diff
@@ -113,6 +113,21 @@ class RabbitMQ {
...
+   * @param  {String} host format: 10.0.0.1:4242
+   * @param  {String} org  github orgId
+   */
+  createDisconnectedJob (host, org) {
+    // we need to send tags for backwards compatibility
+    // also host needs to have http://
+    this.publish('docker.events-stream.disconnected', {
+      host: 'http://' + host,
+      org: org,
+      tags: org
```
Comment (anandkumarpatel) [inline lib/rabbitmq.js]: Do we need to send tags?

**Comment by podviaznikov** [comment]
tested and demoed on gamma

**Pull request merged**